File: frontend/clubbook/dto/AttendanceDto.js
/**
 * AttendanceDto represents the attendance record for a specific class session.
 */
class AttendanceDto {

    /**
    * Creates an instance of AttendanceDto.
    *
    * @param {Date} date - The date of the class session.
    * @param {string} classGroup - The class group for which attendance is recorded.
    * @param {Array<string>} usersIdsAttended - Array of user IDs of students who attended the class.
    * @param {Array<string>} usersIdsNotAttended - Array of user IDs of students who did not attend the class.
    */
    constructor(date, classGroup, usersIdsAttended, usersIdsNotAttended) {
        this.date = date;
        this.classGroup = classGroup;
        this.usersIdsAttended = usersIdsAttended;
        this.usersIdsNotAttended = usersIdsNotAttended;
    }

    /**
     * Validates the attendance record against the class group.
     *
     * @param {Object} classGroup - The class group object containing student information.
     * @returns {boolean} True if the attendance record is valid, otherwise false.
     */
    validate(classGroup) {
        if (this.date === null || this.classGroup === null || this.usersIdsAttended === null || this.usersIdsNotAttended == null) {
            return false;
        }
        if (this.usersIdsAttended.length + this.usersIdsNotAttended.length !== classGroup.students.length) {
            return false;
        }
        return true;
    }
}

export default AttendanceDto;

File: backend/src/main/java/clubbook/backend/dtos/AttendanceDto.java
package clubbook.backend.dtos;

import java.time.LocalDate;
import java.util.List;

/**
 * Data Transfer Object representing the attendance details for a specific class session.
 * Contains information about the date, class group, and lists of users who attended
 * or did not attend the session.
 */
public class AttendanceDto {

    /**
     * The date of the attendance record.
     */
    private LocalDate date;

    /**
     * Identifier for the class group.
     */
    private int classGroup;

    /**
     * List of user IDs who attended the class.
     */
    private List<Integer> usersIdsAttended;

    /**
     * List of user IDs who did not attend the class.
     */
    private List<Integer> usersIdsNotAttended;

    /**
     * Constructs an AttendanceDto with the specified date, class group, and attendance lists.
     *
     * @param date the date of the attendance record
     * @param classGroup the identifier of the class group
     * @param usersIdsAttended the list of user IDs who attended
     * @param usersIdsNotAttended the list of user IDs who did not attend
     */
    public AttendanceDto(LocalDate date, int classGroup, List<Integer> usersIdsAttended, List<Integer> usersIdsNotAttended) {
        this.date = date;
        this.classGroup = classGroup;
        this.usersIdsAttended = usersIdsAttended;
        this.usersIdsNotAttended = usersIdsNotAttended;
    }


    /**
     * Default constructor for AttendanceDto.
     */
    public AttendanceDto() {}

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    public int getClassGroup() {
        return classGroup;
    }

    public void setClassGroup(int classGroup) {
        this.classGroup = classGroup;
    }

    public List<Integer> getUsersIdsAttended() {
        return usersIdsAttended;
    }

    public void setUsersIdsAttended(List<Integer> usersIdsAttended) {
        this.usersIdsAttended = usersIdsAttended;
    }

    public List<Integer> getUsersIdsNotAttended() {
        return usersIdsNotAttended;
    }

    public void setUsersIdsNotAttended(List<Integer> usersIdsNotAttended) {
        this.usersIdsNotAttended = usersIdsNotAttended;
    }
}
